pexec/managed_process.go

```diff
@@ -40,17 +40,46 @@ type ManagedProcess interface {
 // NewManagedProcess returns a new, unstarted, from the given configuration.
 func NewManagedProcess(config ProcessConfig, logger golog.Logger) ManagedProcess {
 	logger = logger.Named(fmt.Sprintf("process.%s_%s", config.ID, config.Name))
+
+	var stopSig syscall.Signal
+	switch config.StopSignal {
+	case "HUP", "SIGHUP", "hangup", "1":
+		stopSig = syscall.SIGHUP
+	case "INT", "SIGINT", "interrupt", "2":
+		stopSig = syscall.SIGINT
+	case "QUIT", "SIGQUIT", "quit", "3":
+		stopSig = syscall.SIGQUIT
+	case "ABRT", "SIGABRT", "aborted", "abort", "6":
+		stopSig = syscall.SIGABRT
+	case "KILL", "SIGKILL", "killed", "kill", "9":
+		stopSig = syscall.SIGKILL
+	case "USR1", "SIGUSR1", "user defined signal 1", "10":
+		stopSig = syscall.SIGUSR1
+	case "USR2", "SIGUSR2", "user defined signal 2", "12":
+		stopSig = syscall.SIGUSR1
+	case "TERM", "SIGTERM", "terminated", "terminate", "15":
+		stopSig = syscall.SIGTERM
+	default:
+		stopSig = syscall.SIGTERM
+	}
+
+	if config.StopTimeout == 0 {
```

Comment:
drop into validate too

Reply:
Same as previous comment, not sure how this can be moved into ProcessConfig.Validate() (if that was what you mean?) because that's a pointer receiver and thus changes the "real" object, causing the robot to think configs have changed and thus reconfigure in an infinite loop. At least as I understand things. Am I missing some pattern that does this correctly? (E.g. sets a default when validating, not just rejects with an error.)